src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs

```diff
@@ -35,6 +36,14 @@ public class BodiesSyncFeed : ActivatedSyncFeed<BodiesSyncBatch?>
 
         private SyncStatusList _syncStatusList;
 
+        private bool ShouldFinish => !_syncConfig.DownloadBodiesInFastSync || AllDownloaded;
+        private bool AllDownloaded => _blockTree.LowestInsertedBodyNumber <= _barrier
+            || WithinOldBarrierDefault;
+
+        // This property was introduced when we switched defaults of barriers from 11052984 to 0 to not disturb existing node operators
+        private bool WithinOldBarrierDefault => _blockTree.LowestInsertedBodyNumber <= DepositContractBarrier
```

Comment:
are you sure it will work correctly in every case?

Reply:
We are covering 99 percent of cases. the only case that is not covered is if someone is syncing, and suddenly stopped the client at exactly around the old default, then unfortunately the sync will stop. other than that, it is all in there

Reply:
Do other network have barrier too?

Reply:
Yeah, they do. The assumption is we only want to do that for mainnet

Reply:
By default we only had barriers on mainnet